WIP: Sc/polytropic 2d wave speed (#1816)
trixi-framework/Trixi.jl: docs/src/multi-physics_coupling.md

# [Multi-physics coupling](@id multi-physics-coupling)

> A complex simulation can consist of different spatial domains in which
> different equations are being solved, different numerical methods being used
> or the grid structure is different.

Reviewer: ```suggestion
A complex simulation can consist of separate spatial domains in which, for
example, different equations are solved, different numerical methods are used,
the mesh structure is different, or some combination.
```

Author: Same as above: coupling is part of a different PR and has been removed from this one.

One example would be a fluid in a tank and an extended hot plate attached to it.
We would then like to solve the Navier-Stokes equations in the fluid domain
and the heat conduction equations in the plate.

> The coupling would happen at the interface through the exchange of thermal energy.

Author: Coupling is part of a different PR and has been removed from this one.

> ## Converter Coupling

Reviewer: ```suggestion
## Converter coupling
```

Author: Same as above: coupling is part of a different PR and has been removed from this one.

> We can have the case where the two systems do not share any variables, but

Reviewer: ```suggestion
It may happen that the two systems to be coupled do not share any variables, but
```

Author: Same as above: coupling is part of a different PR and has been removed from this one.
But I implemented your changes in the coupling PR.

share some of the physics.

> Here, the same physics is just represented in a different form and with
> different variables.

Reviewer: ```suggestion
In such a situation, the same physics is just represented in a different form and with
a different set of variables.
```

Author: Same as above: coupling is part of a different PR and has been removed from this one.

> This is the case for a fluid system on one side and a Vlasov system on the other.

Reviewer: ```suggestion
This is the case, for instance assuming two domains, if there is a fluid system in one domain
and a Vlasov system in the other domain.
```
This is a very specific example and not everyone will necessarily know what a Vlasov system is. Also, the given example elixir couples four different advective equations where the variables "match". It might be worth mentioning more explicitly somewhere that these converter routines need to be hand-rolled by the user within an elixir.

Author: Thanks, I implemented these changes in the coupling PR.

> This is the case for a fluid system on one side and a Vlasov system on the other.

Author: Thanks. I've added this to the coupling PR.

To translate the fields from one description to the other one needs to use
converter functions.

In the general case, we have a system A with ``m`` variables ``u_{A,i}`` and another

> system B with ``n`` variables ``u_{B,j}``.

Reviewer 2: A bit of bikeshedding: I just saw that this is already merged, but as far as I can tell, this should actually be with the double backticks, see [documenter docs](https://documenter.juliadocs.org/stable/man/latex/#Inline-Equations) and https://github.com/trixi-framework/Trixi.jl/pull/1558#discussion_r1370655579. I just checked that we use both version across the docs. I can create a PR that unifies this to use the two backticks version. Or was not any other reason for the suggestion, which I don't see, @DanielDoehring?

> system B with ``n`` variables ``u_{B,j}``.

Reviewer 3: It you could do this, it would be very nice, thanks

> system B with ``n`` variables ``u_{B,j}``.

Reviewer 3: But let's wait for Daniel's input

> system B with ``n`` variables ``u_{B,j}``.

Reviewer 4: The reason I changed this is because (at least as far as I know) documenter can handle both double back ticks and dollar signs for displaying math, but standard markdown does only accept dollar signs, see

$m$ vs ``m``.

So the version with dollar signs should be nicely displayed outside of the doc only, while the version with double back ticks is essentially displayed as code:

https://github.com/trixi-framework/Trixi.jl/blob/09940118196206710d8b3e6ef9abba405701d0f5/docs/src/multi-physics_coupling.md#converter-coupling

> system B with ``n`` variables ``u_{B,j}``.

Reviewer 2: Makes sense. However, it is still not the way how Documenter.jl recommends using LaTeX syntax. I found the [discussion about that in Documenter.jl](https://github.com/JuliaDocs/Documenter.jl/pull/1718#discussion_r733474944).

> system B with ``n`` variables ``u_{B,j}``.

Reviewer 4: Maybe discuss this at Trixi meeting?

> We then define two coupling functions, one that transforms ``u_A`` into ``u_B``

Author: Added this to the coupling PR.

and one that goes the other way.

> In their minimal form they take the position vector ``x`` and state vector ``u``

Author: Added this to the coupling PR.

> and return the transformed variables.

Reviewer: Okay, you partially mention it here. Is this call signature of passing in `x, u, equations_local, equations_neighbor` required under the hood within `SemidiscretizationCoupled`? Is this documented more thoroughly that could be referenced here?
